packages/redis/src/RedisClient.ts

```diff
@@ -214,4 +250,289 @@ export class RedisClient implements IRedisClient, IRequireInitialization, IDispo
             span.finish();
         }
     }
+
+    public async xReadObject<T>(
+        context: SpanContext,
+        type: string | IClassType<T>,
+        streamName: string,
+        id?: string
+    ): Promise<[string, T] | undefined> {
+        const db = this.config.db;
+        const span = this.tracer.startSpan(this.spanOperationName, { childOf: context });
+        this.spanLogAndSetTags(span, this.xReadObject.name, this.config.db, id, streamName);
+        try {
+            const typeName = this.getTypeName(type);
+            const response = await this.client.xread(["block", "0", "streams", streamName, id]);
+            const value = extractStreamValue(response);
+            const streamId = extractStreamId(response);
+            let data;
+            if (value) {
+                const buf = this.config.base64Encode
+                    ? Buffer.from(value, "base64")
+                    : Buffer.from(value);
+                const msg = this.encoder.decode(new Uint8Array(buf), typeName);
+                data = msg.payload;
+            }
+
+            this.metrics.increment(RedisMetrics.XRead, {
+                type,
+                db,
+                result: RedisMetricResults.Success,
+            });
+            return [streamId, data];
+        } catch (e) {
+            failSpan(span, e);
+            this.metrics.increment(RedisMetrics.XRead, {
+                db,
+                result: RedisMetricResults.Error,
+                error: e,
+            });
+            throw e;
+        } finally {
+            span.finish();
+        }
+    }
+
+    public async xGroup(
+        context: SpanContext,
+        streamName: string,
+        consumerGroup: string,
+        supressAlreadyExistsError: boolean = true
+    ): Promise<string> {
+        const db = this.config.db;
+        const span = this.tracer.startSpan(this.spanOperationName, { childOf: context });
+        this.spanLogAndSetTags(span, this.xGroup.name, this.config.db, undefined, streamName);
+        try {
+            const response = await this.client.xgroup([
+                "create",
+                streamName,
+                consumerGroup,
+                "$",
```

Comment:
Relevant context from [the `XGROUP` docs](https://redis.io/commands/xgroup):
> The last argument is the ID of the last item in the stream to consider already delivered. In the above case we used the special ID '$' (that means: the ID of the last item in the stream). In this case the consumers fetching data from that consumer group will only see new elements arriving in the stream.
>
> If instead you want consumers to fetch the whole stream history, use zero as the starting ID for the consumer group...
---
I would argue that clients should be able to influence where in the stream they start consuming from instead of us always defaulting to the end of the stream.

Operationally it can be helpful, for example, to spin up a new consumer from the logical "beginning" of a stream while the existing consumer continues to run in order to introduce breaking changes without downtime.

Reply:
Yeah I agree, I actually had meant to include a config option for this.